android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/NearbyBeaconsFragment.java

```diff
@@ -337,16 +346,48 @@ private void emptyGroupIdQueue() {
 
     for (String groupId : mGroupIdQueue) {
       Log.d(TAG, "groupid " + groupId);
-      pwPairs.add(Utils.getTopRankedPwPairByGroupId(mPwCollection, groupId));
+      pwPairs.add(Utils.getTopRankedPwPairByGroupId(mPwCollection, groupId, mFavorites));
     }
-    Collections.sort(pwPairs, Utils.newDistanceComparator());
+    Collections.sort(pwPairs, new Utils.FavoritesComparator(mFavorites));
     for (PwPair pwPair : pwPairs) {
       mNearbyDeviceAdapter.addItem(pwPair);
     }
     mGroupIdQueue.clear();
     mNearbyDeviceAdapter.notifyDataSetChanged();
   }
 
+  // Check if URL has been favorited
+  public boolean isFavorite(String siteUrl) {
+    return mFavorites.contains(siteUrl);
+  }
+
+  // Toggle URL's favorite status
+  public void flipFavorite(String siteUrl) {
+    if (isFavorite(siteUrl)) {
+      mFavorites.remove(siteUrl);
+      return;
+    }
+    mFavorites.add(siteUrl);
+  }
+
+  private void saveFavorites() {
+    // Write the PW Collection
+    String preferencesKey = getString(R.string.discovery_service_prefs_key);
+    SharedPreferences prefs = getActivity().getSharedPreferences(
```

Comment:
We are trying to use the default prefs now.  Also, just chain these

    getActivity().getDefaultSharedPreferences.edit()
      .putStringSet(...)
      .apply()